src/switch_core_media.c

```diff
@@ -1943,6 +1944,10 @@ static switch_rtp_crypto_mode_t switch_core_session_check_crypto_prefs(switch_co
 		}
 	}
 
+	if (!zstr(val) && (suites = strchr(val, ':'))) {
+		*suites++ = '\0';
```

Comment:
I don't see `suites` used anywhere in this method. I thought it will be used for validation as in https://github.com/team-telnyx/freeswitch/blob/e32cc42acf152d8c23ad37facc5d45882a7b4160/src/switch_core_media.c#L1896

Reply:
The `validation` refers to the function purpose `switch_core_session_check_crypto_prefs` which is to validate and check crypto preference. The value of rtp_secure_media can have this format `<value>[:suites]`. The issue was when dialplan set a specific suite to the rtp_secure_media, something like `true:AES_CM_128_HMAC_SHA1_80` then this specific code will fail
```
        if (!strcasecmp(val, "optional")) {
		return CRYPTO_MODE_OPTIONAL;
	} else if (switch_true(val) || !strcasecmp(val, "mandatory")) {
		return CRYPTO_MODE_MANDATORY;
	} else {
		return CRYPTO_MODE_FORBIDDEN;
	}
```
since `switch_true(val)` cannot handle the parameter suites added in the `true` value thus will return forbidden. The purpose of the `suites` variable is just to split the flag and the suites so I can validate the flag properly. I can validate the suites but doesn't do any advantage, so its better for the suite to be process during SDP negotiation.

Reply:
If you are gonna ask why do we validate and just do the SDP negotiation which the freeswitch vanilla does. The problem is during the RE-INVITE when it causes incompatible destination, according to the RFC the call can still proceed. The problem with the SDP negotiation is that during the processing of SDP, it also modifies the channel vars, flags, audio engine etc and when it fails it doesn't revert those changes. The impact of this will affect the audio when the RE-INVITE fails. To fix it we validate it first and avoid updating the channel audio variables so when it fails it will no longer impact the audio.